drivers/adc/adc_ifx_cat1.c

```diff
@@ -33,15 +35,24 @@ LOG_MODULE_REGISTER(ifx_cat1_adc, CONFIG_ADC_LOG_LEVEL);
 #define ADC_CAT1_RESOLUTION             (12u)
 #define ADC_CAT1_REF_INTERNAL_MV        (1200u)
 
+#if defined(CONFIG_SOC_FAMILY_INFINEON_CAT1B)
+#define IFX_ADC_NUM_CHANNELS 12
```

Comment:
Where does the number '12' come from and can it change for new CAT1B devices in the future? Ideally, rather than assigning IFX_ADC_NUM_CHANNELS a hardcoded value, we should get that value from the appropriate device header file or via DTSI.

Reply:
This is a fair point and should be addressed now.